### ISSUE TITLE
[Bug] Multiple workflows triggered on incomplete data

### ISSUE THREAD

**Issue by humansinstitute** [stakwork/sphinx-tribes-frontend#631]
## Bug

The workflow from ticket: https://github.com/stakwork/sphinx-tribes/issues/1917 is triggered multiple times.

## Context

BUG - This appears to trigger each time new data is loaded on the page. 

I will raise a seperate Bug ticket for this and associate feature to fix. 

- [ ] The workflow is triggered multiple times

Multiple Triggers
![Image](https://github.com/user-attachments/assets/49c00712-e6bf-4f9b-b665-4865c923ebfa)

Each Trigger has increasingly more data loaded which we can see from the data in the vars variable on Stakwork - below: 

1st:
```
    "vars": {
      "productBrief": "Product: . \nProduct Brief: \n* Mission:  \n* Objectives: \n",
      "featureName": null,
      "description": null,
      "examples": [

      ],
      "webhook_url": "people-test.sphinx.chat/features/stories",
      "featureUUID": "csldm4knuku6oosk3rng"
    }
```

2nd:
```
 "vars": {
      "productBrief": "Product: Generate User stories. \nProduct Brief: \n* Mission:  \n* Objectives: \n",
      "featureName": "Generate User stories",
      "description": "Hive Process integrates Stakwork Workflows and LLM based automation into the Hive Development Process. This will reduce workload on Product Managers by streamlining creative tasks and automating the drafting of product artefacts and audit tasks. \n\nIn this feature we will automate the production of User Stories and Requirements from a Product Brief and Feature Brief. \n\nThis will leverage a Stakwork workflow which will take the following steps:\n\n1. Load the product context and specific feature context. \n2. Generate a series of user stories in the workflow as a JSON object. \n3. Iterate through the user stories and score them for relevance against the product and feature context. \n4. Filter out low relevance scores\n5. Return the information to a form in the Hive Process for human review and acceptance.",
      "examples": [

      ],
      "webhook_url": "people-test.sphinx.chat/features/stories",
      "featureUUID": "csldm4knuku6oosk3rng"
    }
```

3rd:
```
"vars": {
      "productBrief": "Product: Generate User stories. \nProduct Brief: \n* Mission:  \n* Objectives: \n",
      "featureName": "Generate User stories",
      "description": null,
      "examples": [

      ],
      "webhook_url": "people-test.sphinx.chat/features/stories",
      "featureUUID": "csldm4knuku6oosk3rng"
    }
```

4th:
```
"vars": {
      "productBrief": "Product: Generate User stories. \nProduct Brief: \n* Mission: Build the delivery methodology and supporting tooling for the AI age. \n \n* Objectives: \n",
      "featureName": "Generate User stories",
      "description": "Hive Process integrates Stakwork Workflows and LLM based automation into the Hive Development Process. This will reduce workload on Product Managers by streamlining creative tasks and automating the drafting of product artefacts and audit tasks. \n\nIn this feature we will automate the production of User Stories and Requirements from a Product Brief and Feature Brief. \n\nThis will leverage a Stakwork workflow which will take the following steps:\n\n1. Load the product context and specific feature context. \n2. Generate a series of user stories in the workflow as a JSON object. \n3. Iterate through the user stories and score them for relevance against the product and feature context. \n4. Filter out low relevance scores\n5. Return the information to a form in the Hive Process for human review and acceptance.",
      "examples": [

      ],
      "webhook_url": "people-test.sphinx.chat/features/stories",
      "featureUUID": "csldm4knuku6oosk3rng"
    }
```

5th 
```
"vars": {
      "productBrief": "Product: Generate User stories. \nProduct Brief: \n* Mission: Build the delivery methodology and supporting tooling for the AI age. \n \n* Objectives: \nObjectives:\n\n1. AI assisted workflows for all PM tasks \n2. Generation of consistent and clean, deliverables tickets \n3. Auto generation of Code and Test Coverage drafts for completion by developers. \n4. Delivery of functional code from a crowd of developers \n5. Self assignment through 'skin in the game' staking for developers \n6. Full developer tooling and environment provision for 0 time to start coding.\n\nTactics:\n\n- Integration via Stakwork for delivery of task via repeatable AI agent workflows. \n- Creation of Graph Database for long term matching of AI Agent success and, learning and refining of process. \n- Provision two sided marketplace for bountying out code tasks to a global developer pool.  \n- Development of a broad and deep bench of expertise in software development for bounty based outsourced tasks. ",
      "featureName": "Generate User stories",
      "description": "Hive Process integrates Stakwork Workflows and LLM based automation into the Hive Development Process. This will reduce workload on Product Managers by streamlining creative tasks and automating the drafting of product artefacts and audit tasks. \n\nIn this feature we will automate the production of User Stories and Requirements from a Product Brief and Feature Brief. \n\nThis will leverage a Stakwork workflow which will take the following steps:\n\n1. Load the product context and specific feature context. \n2. Generate a series of user stories in the workflow as a JSON object. \n3. Iterate through the user stories and score them for relevance against the product and feature context. \n4. Filter out low relevance scores\n5. Return the information to a form in the Hive Process for human review and acceptance.",
      "examples": [

      ],
      "webhook_url": "people-test.sphinx.chat/features/stories",
      "featureUUID": "csldm4knuku6oosk3rng"
    }
  }
```

**Comment by Shoaibdev7** [comment]
.

**Comment by MuhammadUmer44** [comment]
.

**Comment by saithsab877** [comment]
k

**Comment by MahtabBukhari** [comment]
@humansinstitute Could you please assign me?

**Comment by humansinstitute** [comment]
I just tried a quick test on Staging and still got three request to stakwork on the backend. 

We can see the request being posted here too: 

https://www.loom.com/share/854c756290b54e47951b06c22745c631?sid=049ceca6-f383-4f1d-9abf-4bcdea5a18c4

![Image](https://github.com/user-attachments/assets/00810c52-202d-49b9-b7e4-dad0dbbc3988)

Can we take a different tack here. 

1. Initial when we loaded the modal we displayed the full object. Lets do that and display that in the modal
2. Add a send button to the modal
3. On click of the send button - post the request to stakwork
4. On ack response from stakwork - display the response in the modal (as we do now)
5. On click of cancel - do not send and exit the modal

@MahtabBukhari does that approach make sense? I think this will be simpler as I thin there are a few load processes in the background which are tripping us up trying to do this "onLoad" so lets make it a human decision, once we can confirm the data object is correct.
